OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationController.java

```diff
@@ -58,7 +58,18 @@
    private static boolean locationCoarse;
 
    static final Object syncLock = new Object() {};
-   static LocationHandlerThread locationHandlerThread;
+
+   private static LocationHandlerThread locationHandlerThread;
+   static LocationHandlerThread getLocationHandlerThread() {
+      if (locationHandlerThread == null) {
```

Comment:
Removed redundant `if` outside of synchronized

Reply:
actually this was discussed on https://github.com/OneSignal/OneSignal-Android-SDK/pull/1092#discussion_r461171325

Reply:
I see, make sense now!